components/omega/doc/design/PGrad.md

```diff
@@ -0,0 +1,224 @@
+
+$$
+T^p_{e,k} = -\left[ \alpha_{i,k} \right]_e \nabla p_{i,k} - \nabla \Phi_{i,k},
+$$
+
+where the second term is necessary to account for tilted layers that occur when using a general vertical coordinate.
+In this equation, $\alpha_{i,k}$ is the specific volume for cell $i$ at the mid-point of level $k$, $p_{i,k}$ is the pressure, and $\Phi_{i,k}$ is the geopotential.
+The discrete gradient operator at an edge is:
+
+$$
+ \nabla {(\cdot)} = \frac{1}{d_e} \sum_{i\in CE(e)} -n_{e,i} (\cdot)_i
+$$
+
+where $d_e$ is the distance between cell centers, $CE(e)$ are the cells on edge $e$, and $n_{e,i}$ is the sign of the edge normal with respect to cell $i$.
+The horizontal averaging operator is:
+
+$$
+ [\cdot]_e = \frac{1}{2}\sum_{i\in CE(e)} (\cdot)_i
+$$
+
+Therefore, the centered pressure gradient will be calculated as:
+
+$$
+ T^p_{e,k} = \frac{1}{d_e} \left( [\alpha_{i,k}]_e \sum_{i \in CE(e)} n_{e,i} p_{k,i} + \sum_{i\in CE(e)} n_{e,i} \Phi_{k,i}\right),
+$$
+
+$$
+ = \frac{1}{d_e} \left(  \sum_{i \in CE(e)} n_{e,i} \left( [\alpha_{i,k}]_ep_{k,i} + \Phi_{k,i} \right) \right),
+$$
+
+### 3.2 Barotropic Pressure Gradient
+
+When split baroclinic-barotropic time stepping is implemented in the future, the barotropic pressure gradient will be calculated by the pressure gradient class.
+The barotropic pressure gradient is found by depth integrating the pressure gradient.
+The pressure is
+
+$$
+p(z) = p_b - g \int^z_{-h} \rho dz^\prime,
+$$
+
+where $p_b$ is the bottom pressure.
+The bottom pressure is the sum of the atmospheric surface pressure, $p_s$, and the pressure contribution of the water column:
+
+$$
+p(z) &=  p_s + g\int_{-h}^\eta \rho dz - g \int^z_{-h} \rho dz^\prime, \\
+     &=  p_s + g\rho_0\widetilde{H} - g \int^z_{-h} \rho dz^\prime,
+$$
+
+where the total water column pseudo height is expressed by
+
+$$
+\widetilde{H} = \int_{-h}^\eta \frac{\rho}{\rho_0} dz.
+$$
+
+$\widetilde{H}$ is the prognositc variable in the barotropic continuity equation.
+The vertical integral of the pressure gradient is
+
+$$
+\frac{1}{\rho_0\widetilde{H}}\int^\eta_{-h} \nabla p dz &= \frac{1}{\rho_0\widetilde{H}}\int^\eta_{-h} \nabla \left( p_s + g\rho_0 \widetilde{H} - g \int^z_{-h} \rho dz^\prime \right) dz, \\
+                           &= \frac{H}{\rho_0\widetilde{H}}\nabla p_s + \frac{gH}{\widetilde{H}}\nabla \widetilde{H} - \frac{g}{\rho_0\widetilde{H}} \int_{-h}^\eta \left( \nabla \int_{-h}^z \rho dz^\prime\right) dz,
+$$
+
+where the height of the water column is represented by $H$.
+The $1/\rho_0\widetilde{H}$ factor comes vertically integrating the material derivative and expressing the resulting barotropic momentum equation in non-conservative form.
+
+Therefore, the barotorpic pressure gradient term is discretized as:
+
+$$
+\overline{T}_e^p = g\left[ \frac{H_i}{\widetilde{H}_i} \right]_e\sum_{i \in CE(e)} n_{e,i}\widetilde{H}_e
```

Comment:
How will we know $H_i$ in this equation?  This seems pretty unknowable in the barotropic equation to me.

Reply:
I've decided to leave the barotropic pressure gradient to be discussed in the split timestepping design doc.